account_asset_maintenance/models/account_asset_asset.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountAssetAsset(models.Model):
+    _inherit = 'account.asset.asset'
+
+    equipment_ids = fields.Many2many(
+        comodel_name="maintenance.equipment", string="Equipments",
+    )
+    equipment_count = fields.Integer(
+        string="Equipment count", compute="_compute_equipment_count",
+    )
+
+    @api.depends('equipment_ids')
+    def _compute_equipment_count(self):
+        for asset in self:
+            asset.equipment_count = len(asset.equipment_ids)
+
+    @api.multi
+    def button_open_equipment(self):
```

Comment:
Wouldn't it be nicer to directly open the form view if there is only one record to show.